### PR TITLE
1160: add exact amount display method

### PR DESCRIPTION
**Pull request by lauchaves**
### Description 

- This PR fix issue [#1160 ](https://github.com/near/near-sdk-rs/issues/1160)from [near-sdk ]( https://github.com/near/near-sdk-rs/)

- Need to add the `exact_amount_display` method to format `yoctoNEAR` values for clearer error messages in `near-sdk` storage management.

### Implementation 

- Added `exact_amount_display` Method: it formats NearToken amounts for display as yoctoNEAR or NEAR.
- Based on [exact-display implementation from near-cli-rs](https://github.com/near/near-cli-rs/blob/7fca81bb5544b40743290d3a951c1f1ff31a95d9/src/types/near_token.rs#L18-L35) 
- Added tests for various cases: zero value, small values, exact NEAR values, and fractional NEAR values.

### Example Usage: 
```
let required_cost = NearToken::from_yoctonear(15 * 10_u128.pow(23));
format!("Must attach {} to cover storage", required_cost.exact_amount_display())
```

### CODE REVIEW

**Comment by lauchaves** [comment]
@frol this one is ready for review!

**Review by akorchyn — APPROVED** [review]
LGTM

I'd like to invite you to participate in the [Race of Sloths](https://race-of-sloths.com/). Just mention @race-of-sloths user in your GitHub comment or PR description to join!

**Review by frol — APPROVED** [review]
@lauchaves Great!

**Comment by lauchaves** [comment]
Hey @frol, @akorchyn, I didn't know about the Race of Sloths! I just joined, that's super cool! Thanks so much for the invite! 🦥🚀

**Comment by lauchaves** [comment]
@race-of-sloths pls, include my pr in the race!

**Comment by race-of-sloths** [comment]
@lauchaves Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
New Sloth joined the Race! Welcome!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/lauchaves?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/lauchaves?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/lauchaves)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/lauchaves)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Comment by race-of-sloths** [comment]
### 🔄 The PR has been merged.
>[!IMPORTANT]
This pull request is a part of the Race of Sloths and has not been scored yet. Scoring will close in 24 hours! 🕰️

**Comment by frol** [comment]
@race-of-sloths score 2

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@frol, thank you for scoring this pull request in the Race of Sloths!

**Comment by race-of-sloths** [comment]
### ✅ PR is finalized! 
Your contribution is much appreciated with a final score of 2!
You have received 30 (20 base + 10 weekly bonus) Sloth points for this contribution


Another weekly streak completed, well done @lauchaves! To keep your weekly streak and get another bonus make pull request next week! Looking forward to see you in race-of-sloths

**Pull request merged**
